make_wheels.py

```diff
@@ -71,18 +71,18 @@ def write_wheel(out_dir, *, name, version, tag, metadata, description, contents)
     dist_info  = f'{name}-{version}.dist-info'
     return write_wheel_file(os.path.join(out_dir, wheel_name), {
         **contents,
-        f'{dist_info}/METADATA': make_message({
-            'Metadata-Version': '2.1',
-            'Name': name,
-            'Version': version,
-            **metadata,
-        }, description),
-        f'{dist_info}/WHEEL': make_message({
-            'Wheel-Version': '1.0',
-            'Generator': 'ziglang make_wheels.py',
-            'Root-Is-Purelib': 'false',
-            'Tag': tag,
-        }),
+        f'{dist_info}/METADATA': make_message([
+            ('Metadata-Version', '2.4'),
+            ('Name', name),
+            ('Version', version),
+            *metadata,
+        ], description),
+        f'{dist_info}/WHEEL': make_message([
+            ('Wheel-Version', '1.0'),
+            ('Generator', 'ziglang make_wheels.py'),
+            ('Root-Is-Purelib', 'false'),
+            ('Tag', tag),
+        ]),
```

Comment:
This metadata version bump is backwards compatible. Please see https://peps.python.org/pep-0639/#backwards-compatibility for more. I tested recent versions of `pip` and `uv`, and they installed a wheel I generated for my M-series device without hiccups. I'll need to see if compliance with PyPI uploads is working, though, that's more important to figure out.